### PR TITLE
[Bugfix] Remove xformers requirement for Pixtral

### PR DESCRIPTION
**Pull request by mgoin**
This PR adds a torch SDPA implementation for PixtralHF attention 

FIX https://github.com/vllm-project/vllm/pull/9036#issuecomment-2428076474
> I'm trying to run LlavaNextForConditionalGeneration on a non-CUDA hardware platform, but when I import llava_next, it pulls in pixtral, which in turn imports xformers. From my understanding, xformers is required only for CUDA support. Is there a way to avoid this dependency or run LlavaNextForConditionalGeneration without xformers on non-CUDA platforms?

### CODE REVIEW

File: vllm/model_executor/models/pixtral.py [hunk 1]
```diff
@@ -827,12 +835,32 @@ def forward(
...
+
+            # upcast attention to fp32
+            attn_weights = nn.functional.softmax(attn_weights,
+                                                 dim=-1,
+                                                 dtype=torch.float32).to(
+                                                     q.dtype)
+            v = v.view(batch, patches, self.n_heads,
+                       self.head_dim).transpose(1, 2)
+            out = torch.matmul(attn_weights, v)
+            out = out.transpose(1, 2)
```
[the remark this answers is not in the document]
Comment (mgoin) [reply vllm/model_executor/models/pixtral.py:862]: I tried using nn.functional.scaled_dot_product_attention but got gibberish for output, need to understand further

[the remark this answers is not in the document]
Comment (DarkLight1337) [reply vllm/model_executor/models/pixtral.py:862]: See if you can fix this quickly, If not, we can merge this first as it fixes https://github.com/vllm-project/vllm/pull/9036#issuecomment-2428076474

**Review by DarkLight1337 — APPROVED** [review]

File: vllm/model_executor/models/pixtral.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (DarkLight1337) [reply vllm/model_executor/models/pixtral.py:862]: Quick ping, in case you forgot about this.

[the remark this answers is not in the document]
Comment (mgoin) [reply vllm/model_executor/models/pixtral.py:862]: I didn't make any progress on this, and then I tried `out = out.transpose(1, 2)` 😞 all good now with sdpa, pushing

**Pull request merged**